### PR TITLE
feat(op-challenger): Responder Step Function

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Adds a `Step` function to the `Responder` interface.

**TODO**: Implement the `Step` function for the `faultResponder` to pack the step function call and execute the tx.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6221** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6221" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6221?utm_source=stack-comment).

File: op-challenger/fault/types.go [hunk 1]
```diff
@@ -12,6 +12,15 @@ var (
 	ErrIndexTooLarge = errors.New("index is larger than the maximum index")
 )
 
+// StepData encapsulates the data needed to perform a step.
+type StepData struct {
```
Comment (trianglesphere) [inline op-challenger/fault/types.go]: naming conflict

**Review by trianglesphere — COMMENTED** [review]
Naming conflict but otherwise lgtm

File: op-challenger/fault/types.go (hunk 1 above)
Comment (refcell) [reply op-challenger/fault/types.go]: Ah yep I just changed 👍

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**